### PR TITLE
Issue 14 hide author option

### LINKED ISSUE #14: Hide author option
Add an option for author names to be hidden from students.
Add an option for author names to be hidden from teachers.
Reload page to see changes applied.

### PR DESCRIPTION
**Pull request by l3n0ire**
## Details

- Added options in add-board and configuration-modal to hide author names from students and from teachers
- When "show author names" is unchecked, author names will appear as Anonymous to other students

Closes #14

### CODE REVIEW

**Review by JoelWiebe — CHANGES_REQUESTED** [review]
Adding separate hide options for teacher and students is a good idea. 

Adding these to the permissions tab makes sense. Good labels for each option.

For some reason, as a teacher, the canvas continues to show Anonymous on my old posts, but when I open them up I can see the author (i.e., see the Test 1 post below). Permission are set so that both students and teachers should be able to see the author names. Any idea why this might be happening? 

image

**Comment by l3n0ire** [comment]
So when the canvas loads it uses PostService(which gets the posts from firebase) to get the existing fabric objects and renders them. So somehow the fabric js object for Test1 and Test2 had the author text field saved as Anonymous, but the actual author is still correct (ex JoelWeibe).  So thats why the author name shows up in the expanded post modal. I could add in a function to update the author text fields to reflect the permission. Right now it only updates to anonymous when "show author name" is unchecked, but doesn't do anything if "show author name" is checked

**Comment by JoelWiebe** [comment]
Thanks @l3n0ire, that sounds like a great idea to update author names when show author name is checked, to ensure all author names are shown again after being anonymized. Let's try that, then I'll do a quick review again.

**Comment by Ammar-T** [comment]
I tried hiding author names for teachers, however the names don't become "Anonymous" on the fabric post until I refresh the page. Could you try to update it so the change takes effect immediately without having to refresh? @l3n0ire

File: src/app/components/canvas/canvas.component.ts [hunk 1]
```diff
@@ -88,6 +89,9 @@ export class CanvasComponent {
           board.permissions.allowStudentMoveAny ? this.lockPostsMovement(false) : this.lockPostsMovement(true)
           board.bgImage ? this.updateBackground(board.bgImage.url, board.bgImage.imgSettings) : null
           this.updateShowAddPost(this.board.permissions)
+          if(!((this.user.role =="student" && this.board.permissions.showAuthorName) || this.user.role =="teacher")){
```
Comment (Ammar-T) [inline src/app/components/canvas/canvas.component.ts]: Can you extract this logic into smaller chunks, similar to my suggestions in the locking PR?

**Comment by JoelWiebe** [comment]
Great work!

Tested:
- Teacher toggles anonymity of authors for teachers
- Teacher toggles anonymity of authors for students

Logged in on two teacher accounts and one student account

- When T1 made a change to anonymity for teachers, effect was immediate for T1, but T2 had to refresh their browser for changes to take effect
- When T1 made a change to anonymity for students, S1 how to refresh their browser for changes to take effect

Is it easy to make changes take immediate effect for other students and teachers too? If this is a bit of work, please don't worry, this should be quite sufficient for the time being as is!

**Review by JoelWiebe — APPROVED** [review]

**Comment by JoelWiebe** [comment]
@StevenWuzz can you provide your review (accept/request changes) when you get a chance?

**Comment by l3n0ire** [comment]
> Great work!
> 
> Tested:
> 
> * Teacher toggles anonymity of authors for teachers
> * Teacher toggles anonymity of authors for students
> 
> Logged in on two teacher accounts and one student account
> 
> * When T1 made a change to anonymity for teachers, effect was immediate for T1, but T2 had to refresh their browser for changes to take effect
> * When T1 made a change to anonymity for students, S1 how to refresh their browser for changes to take effect
> 
> Is it easy to make changes take immediate effect for other students and teachers too? If this is a bit of work, please don't worry, this should be quite sufficient for the time being as is!

@JoelWiebe I'll create a separate issue for this and work on it this week

**Review by Ammar-T — APPROVED** [review]
💯

**Review by StevenWuzz — APPROVED** [review]

**Comment by JoelWiebe** [comment]
Thanks everyone!!

@l3n0ire Please feel free to Resolve Conflicts and Merge :)

**Pull request merged**